execution/gethexec/express_lane_service.go

```diff
@@ -341,35 +354,52 @@ func (es *expressLaneService) sequenceExpressLaneSubmission(
 		return timeboost.ErrSequenceNumberTooLow
 	}
 	// Check if a duplicate submission exists already, and reject if so.
-	if _, exists := es.messagesBySequenceNumber[msg.SequenceNumber]; exists {
+	if _, exists := es.msgAndResultBySequenceNumber[msg.SequenceNumber]; exists {
 		return timeboost.ErrDuplicateSequenceNumber
 	}
 	// Log an informational warning if the message's sequence number is in the future.
 	if msg.SequenceNumber > control.sequence {
 		log.Info("Received express lane submission with future sequence number", "SequenceNumber", msg.SequenceNumber)
 	}
 	// Put into the sequence number map.
-	es.messagesBySequenceNumber[msg.SequenceNumber] = msg
+	resultChan := make(chan error, 1)
+	es.msgAndResultBySequenceNumber[msg.SequenceNumber] = &msgAndResult{msg, resultChan}
 
-	for {
+	now := time.Now()
+	for es.roundTimingInfo.RoundNumber() == msg.Round { // This is an important check that mitigates a security concern
```

Comment:
This check ensures that the controller for this round is not allowed to send transactions once the next round starts, which would be possible without this check by just queuing up future seq number transactions and then send the transaction with unblocking seq num right at the end of the round.

I'll update the comment to include more info

Reply:
It should be checking for msg.Round because only the current incoming tx (msg) should be able to unblock future txs from the map and all of them should share the same round number anyway (if not they'd not pass the initial validation)